hack/xref-helpmsgs-manpages

```diff
@@ -0,0 +1,334 @@
+    # This is deliberate: it's because Getopt::Long only operates on @ARGV
+    # and there's no clean way to make it use @_.
+    handle_opts();                      # will set package globals
+
+    # Fetch command-line arguments.  Barf if too many.
+    die "$ME: Too many arguments; try $ME --help\n"                 if @ARGV;
+
+    my $help = skopeo_help();
+    my $man  = skopeo_man('skopeo');
+
+    xref_by_help($help, $man);
+    xref_by_man($help, $man);
+
+    exit !!$Errs;
+}
+
+###############################################################################
+# BEGIN cross-referencing
+
+##################
+#  xref_by_help  #  Find keys in '--help' but not in man
+##################
+sub xref_by_help {
+    my ($help, $man, @subcommand) = @_;
+
+    for my $k (sort keys %$help) {
+        if (exists $man->{$k}) {
+            if (ref $help->{$k}) {
+                xref_by_help($help->{$k}, $man->{$k}, @subcommand, $k);
+            }
+            # Otherwise, non-ref is leaf node such as a --option
+        }
+        else {
+            my $man = $man->{_path} || 'man';
+            warn "$ME: skopeo @subcommand --help lists $k, but $k not in $man\n";
+            ++$Errs;
+        }
+    }
+}
+
+#################
+#  xref_by_man  #  Find keys in man pages but not in --help
+#################
+#
+# In an ideal world we could share the functionality in one function; but
+# there are just too many special cases in man pages.
+#
+sub xref_by_man {
+    my ($help, $man, @subcommand) = @_;
+
+    # FIXME: this generates way too much output
+    for my $k (grep { $_ ne '_path' } sort keys %$man) {
+        if (exists $help->{$k}) {
+            if (ref $man->{$k}) {
+                xref_by_man($help->{$k}, $man->{$k}, @subcommand, $k);
+            }
+        }
+        elsif ($k ne '--help' && $k ne '-h') {
+            my $man = $man->{_path} || 'man';
+
+            # Special case: skopeo-inspect serves dual purpose (image, ctr)
+            my %ignore = map { $_ => 1 } qw(-l -s -t --latest --size --type);
+            next if $man =~ /-inspect/ && $ignore{$k};
+
+            # Special case: skopeo-diff serves dual purpose (image, ctr)
+            my %diffignore = map { $_ => 1 } qw(-l --latest );
+            next if $man =~ /-diff/ && $diffignore{$k};
+
+            # Special case: the 'trust' man page is a mess
+            next if $man =~ /-trust/;
+
+            # Special case: '--net' is an undocumented shortcut
+            next if $k eq '--net' && $help->{'--network'};
+
+            # Special case: these are actually global options
+            next if $k =~ /^--(cni-config-dir|runtime)$/ && $man =~ /-build/;
+
+            # Special case: weirdness with Cobra and global/local options
+            next if $k eq '--namespace' && $man =~ /-ps/;
+
+            next if "@subcommand" eq 'system' && $k eq 'service';
+
+            # Special case: skopeo completion is a hidden command
+            next if $k eq 'completion';
```

Comment:
This looks (mostly?) Podman-specific.

(I have no idea how to sanely maintain the script across repositories and prevent drift, and this script is already very useful even with these exceptions, so it well might be worth keeping it as is to decrease the maintenance overhead.)

Reply:
I'm sorry! This is part of the WIP: I intend to strip this and other such sections before removing the WIP.

As you identified, maintaining across repositories is a no-hope situation.

Reply:
> I have no idea how to sanely maintain the script across repositories and prevent drift

> As you identified, maintaining across repositories is a no-hope situation.

FWIW: The way this is done with other shared scripts is a bit complex, but functional and reliable:  Stick the script into c/automation (along with it's unit tests) so it installs directly into the VMs via c/automation_images.  Then call the script from the task.  Stability is maintained by tagging "releases" of c/automation, and [pinning that version into the VM build process](https://github.com/containers/automation_images/blob/main/lib.sh#L27).

I will admit it's a lot of work up-front, but I think it does pay off long-term by not having weird problems happening on release branches (where VM images don't ever change).

Reply:
If the question is "can we share a version of this script between podman, buildah, and skopeo", the answer is "not in any sane manner". Each repo has different conventions and special cases. Maintaining this in a share repo would mean that every edit would need to be tested in podman/buildah/skopeo ... and we know that will not happen.

History has shown that these scripts are fine as they are, and do not get new-functionality edits, only new-special-case-handling edits (see git log in podman and buildah). As much as I hate duplication, I think it's the most reasonable approach in this case.

Reply:
It seems vaguely possible to share the core of the logic, and then either have a shared script that has the special cases conditioned on the program under check (like the `podman-image-trust` case that came up earlier, with that name), or just have the special cases maintained directly in the individual Podman/Skopeo/… repos.

OTOH now that the core logic works, it seems quite unlikely to need substantial changes, so the payoff of sharing _that_ part might well be absolutely not worth the effort to share it.

Reply:
Yes I agree. It only makes sense if the core logic is substantial, and important enough to have it's own unit-tests.  I'm not familiar with this code at all, so I trust Ed's judgement.  My intention was simply to point out the possibility in case it made sense.